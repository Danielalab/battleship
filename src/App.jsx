import './App.css';
import { Routes, Route } from 'react-router-dom';
import BattleshipView from './views/BattleshipView';

function App() {
  return (
    <Routes>
      <Route path="/" element={<BattleshipView />} />
    </Routes>
  );
}

export default App;
