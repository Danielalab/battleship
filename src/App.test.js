import 'jest-styled-components';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import { render } from '@testing-library/react';
import App from './App';

describe('Buttons', () => {
  test('Should renders Button component', () => {
    /* render component with router */
    const history = createMemoryHistory();
    const { container } = render(
      <Router location={history}>
        <App />
      </Router>,
    );
    expect(container).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
