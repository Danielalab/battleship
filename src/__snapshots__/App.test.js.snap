// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Buttons Should renders Button component 1`] = `
.c4 {
  cursor: pointer;
  font-size: 1.5rem;
  font-family: 'Press Start 2P',cursive;
  margin: 0.75rem;
  padding: 0.5rem 1rem;
  text-transform: uppercase;
}

.c8 {
  background-color: #113CFC;
  color: #ffffff;
}

.c8:disabled {
  background-color: transparent;
  color: #ffffffa2;
}

.c2 {
  color: #05FC32;
  font-size: 2rem;
  margin: 0 0.75rem;
}

.c3 {
  background: transparent;
  color: #ffffff;
  font-size: 1.5rem;
  font-family: 'Press Start 2P',cursive;
  margin: 0 0.75rem;
  padding: 0.5rem 1rem;
  outline: 0;
}

.c7 {
  background: transparent;
  color: #ffffffa2;
  font-size: 1.5rem;
  font-family: 'Press Start 2P',cursive;
  margin: 0 0.75rem;
  padding: 0.5rem 1rem;
  outline: 0;
}

.c5 {
  background-color: #FCC82B;
  color: #000000;
}

.c6 {
  background-color: transparent;
  color: #ffffffa2;
}

.c1 {
  margin: 2rem 0;
}

.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100vh;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <label
        class="c2"
        for="user-nickname"
      >
        Nickname:
      </label>
      <input
        class="c3"
        id="user-nickname"
        placeholder="Jack Sparrow"
        type="text"
        value=""
      />
    </div>
    <div
      class="c1"
    >
      <button
        class="c4 c5"
      >
        easy
      </button>
      <button
        class="c4 c6"
      >
        medium
      </button>
      <button
        class="c4 c6"
      >
        hard
      </button>
      <button
        class="c4 c6"
      >
        choose my chances
      </button>
    </div>
    <div
      class="c1"
    >
      <label
        class="c2"
        for="user-chances"
      >
        Chances:
        <input
          class="c7"
          id="user-chances"
          min="20"
          readonly=""
          type="text"
          value="Infinity"
        />
      </label>
    </div>
    <button
      class="c4 c8"
      disabled=""
      type="submit"
    >
      start game
    </button>
  </div>
</div>
`;
